### ISSUE TITLE
CSS Modules no longer work

### ISSUE THREAD

**Issue by Cinamonas** [nozzle/react-static#601]
As far as I can tell, #574 may have broken the usage of CSS Modules in React-Static. I’ve upgraded to the latest version and noticed that `import styles from 'Foo.scss'` no longer works as expected. If you attempt to reference a class name on the `styles` object, it results in `undefined`.

### Environment

1.  `react-static -v`: 5.8.2 (first non-working release, last working release is 5.8.1)
2.  `node -v`: 8.9.3
3.  `yarn --version or npm -v`: 5.6.0
4.  Operating system: macOS 10.13.4

### Steps to Reproduce the problem

1. Go to [examples/sass](https://github.com/nozzle/react-static/tree/master/examples/sass)
2. `npm i`
3. `npm i react-static@5.8.2` (first non-working release)
4. Open `static.config.js` and add `modules: true` option on `css-loader`
5. Open `App.js` and replace `import './app.scss'` with `import styles from './app.scss'`
6. Add `console.log(styles.content)` below the `app.scss`  import
7. Run `npm stage`
8. Notice that the `console.log` call results in `undefined`

### Expected Behavior

If you did the steps outlined above on 5.8.1, it would work. You’d just need to remove the `if` condition surrounding `ExtractTextPlugin.extract` call in `static.config.js` (which was introduced to conform to changes in #574).

---

Let me know if I’m missing some trivial configuration that would allow me to use CSS Modules. Otherwise, changes in #574 need to be revisited.

/cc @lottamus

**Comment by JustFly1984** [comment]
I already pointed out that it is due to webpack.config.js refactoring, and introducing boolean isNode

**Comment by lottamus** [comment]
It looks like we just need to add `modules: true` to enable CSS modules

https://javascriptplayground.com/css-modules-webpack-react/

**Comment by Cinamonas** [comment]
@lottamus, as pointed out above, I’m adding that. My configuration hasn’t changed, but the CSS Modules no longer work after 5.8.2.

**Comment by rohmanhm** [comment]
Also, I can't get CSS Code Splitting on my app.
The CSS bundles now just compiled into a single CSS file.

**Comment by sudarshang** [comment]
I have a PR which upgrades react-static to webpack 4. Do you want to check if this issue is resolved by using that branch?

https://github.com/nozzle/react-static/pull/605

**Comment by zackljackson** [comment]
https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/tree/webpack-4

`npm I extract-css-chunks-webpack-plugin@next`

@Cinamonas @rohmanhm

**Comment by zackljackson** [comment]
Plugin now released for webpack 4

**Comment by mrchief** [comment]
Are we using wp4 in react-static?

**Comment by Cinamonas** [comment]
@zackljackson, could you elaborate? I don’t quite understand how this is a fix to the CSS Modules issue I described.

**Comment by JustFly1984** [comment]
Now, since webpack 4 is implemented, is there a guide how to enable CSS modules? Both of my projects made with React-static heavily depends on CSS modules, an it CSS modules are broken since 5.8.1. I want to update both projects to latest version, since NPM has found a ton of vulnerabilities in 5.8.1 but I can’t update until I’m sure that CSS modules are working with webpack 4

**Comment by JustFly1984** [comment]
Is there any updates or estimates?

**Comment by mrchief** [comment]
I got the CSSModules working with old setup. Had to remove the ExtractCssChunks part. Here's my working cut with `5.9.7` and `etwp @ ^3.0.2`

```
import ExtractCssPlugin from 'extract-text-webpack-plugin'
import path from 'path'
import postcssFlexbugsFixes from 'postcss-flexbugs-fixes'
import postcssPresetEnv from 'postcss-preset-env'

export default ({ config, stage }) => {
  let cssLoader = [
    {
      loader: 'css-loader',
      options: {
        localIdentName:
          stage === 'prod' ? '[hash:base64:8]' : '[path][name]---[local]',
        minimize: stage === 'prod',
        sourceMap: stage === 'dev',
        url: false
      }
    },
    {
      loader: 'postcss-loader',
      options: {
        // Necessary for external CSS imports to work
        // https://github.com/facebookincubator/create-react-app/issues/2677
        sourceMap: true,
        ident: 'postcss',
        plugins: () => [
          postcssFlexbugsFixes,
          require('postcss-import')(),
          require('postcss-url')(),
          postcssPresetEnv({
            features: {
              'nesting-rules': true
            }
          })
        ]
      }
    }
  ]

  if (stage === 'dev') {
    cssLoader = ['style-loader'].concat(cssLoader)
  } else if (stage === 'prod') {
    cssLoader = ExtractCssPlugin.extract({
      fallback: {
        loader: 'style-loader',
        options: {
          sourceMap: false,
          hmr: false
        }
      },
      use: cssLoader
    })
  }

  return {
    test: /\.css$/, 
    loader: cssLoader
  }
}
```

**Comment by ScriptedAlchemy** [comment]
Does this help?

Looks related. Maybe?
https://github.com/webpack/webpack/issues/1754

I’m relocating to the west coast so I’m pretty much offline till after the 4th :)

**Comment by kurdin** [comment]
After more time spending on the problem and tests, looks like this issue not related to version @5.9.7, at least for me. 

This is more global vs local install issue, so I solved this by installing react-static locally and using it via `node_modules/.bin/react-static`

**Comment by ScriptedAlchemy** [comment]
hey guys, sorry for such a delay.

Regarding my css loader `extract-css-chunks-webpack-plugin` which is part of react static. 
Im about to release a next tag build on npm which should get HMR and CSS modules working again.

**Comment by JustFly1984** [comment]
Thanx, that means I can try to update! Can't wait!!!

**Comment by ScriptedAlchemy** [comment]
Soon as it’s merged I’ll update the project. Otherwise just manually
install the latest into the repo

**Comment by ScriptedAlchemy** [comment]
@JustFly1984 This has been merged. @tannerlinsley i think you might be able to close the issue if @Cinamonas can confirm

**Comment by Cinamonas** [comment]
I’ll upgrade to v6 as soon as it’s out of beta.

**Comment by tannerlinsley** [comment]
Is there anything else that needs to happen in v6 to make css modules work? I'll be honest, I don't use them (all emotion/styled-components here), so I'm not sure how it all works haha.

**Comment by jakeFeldman** [comment]
Is there any updates here? I have CSS modules working for dev in version 5.9.7, but when I build a production version the css is not extracted into the markup. I tried to migrate to v6 but I was not able to build/start the project being tracked in #685. Thanks in advance.

**Comment by blakeperdue** [comment]
I'm including sass files (via `import './app.scss'` in App.js) and I'm running RS6 (6.0.0-beta.13) and `extract-css-chunks-webpack-plugin@next` (3.1.0-beta.2) but it's not working for me. I'm also using Material-UI (1.4.0).

Maybe I'm missing something. I'm including links to my source files in case that's helpful:

`static.config.js`: https://cl.ly/0r2I3O3u260u
`index.js`:  https://cl.ly/2s2z3L0u232p
`App.js`: https://cl.ly/3E091G0N3H1W
`Document.js`: https://cl.ly/1G3W200H2E03
`package.json`: https://cl.ly/0E1W412A0y1a

When I build or just run local dev, you can see all the Material UI styles are properly getting loaded and applied. However, none of the styles in the `app.scss` file are being included anywhere:

![image](https://user-images.githubusercontent.com/1642071/43038660-732fe394-8ceb-11e8-93b4-49228c3c4695.png)

@ScriptedAlchemy I know you're busy, but would love if you could just point me in the right direction. That would be super helpful. Thanks!

**Comment by blakeperdue** [comment]
I should also mention I do *not* have a `plugins` folder with any plugin files. I saw that the `sass` example did have this but other examples that use plugins did not. Also, the link in this repo for `react-static-plugin-sass` plugin is returning 404:

https://github.com/nozzle/react-static-plugin-sass

Not sure if that plugin has been deprecated or what. Any info would be appreciated!

**Comment by blakeperdue** [comment]
I'm guessing I *do* need to include this file and put it in the plugins root folder:
https://github.com/nozzle/react-static/blob/master/examples/sass/plugins/react-static-plugin-sass.js

I'm not sure if that file needs modifications though?

**Comment by tannerlinsley** [comment]
The sass plugin is currently not published yet an is still being developed
in the sass example’s plugin directory.

**Comment by blakeperdue** [comment]
@tannerlinsley ahh, thanks for the clarification. Sorry the flurry of comments. I suppose I'll just stick to CSS in JS until that sass example & plugin is finished. Thanks.

**Comment by JustFly1984** [comment]
I've managed to setup CSS Modules in development mode with react-static 6 beta-16, but prod build fails.

**Comment by JustFly1984** [comment]
I fixed prod build, but not CSS modules. I'm debugging react-static's webpack configs, and logging out configs:

```
=> Bundling App...
stage:  prod
config.module.rules[1].oneOf:  [{ 
    test: /\.(js|jsx)$/,
    exclude: /(node_modules|\/Users\/justfly\/projects\/aggrow-static\/node_modules)/,
    use: [ [Object] ] 
}, { 
    test: /\.css$/,
    loader: [ 
      '/Users/justfly/projects/aggrow-static/node_modules/extract-css-chunks-webpack-plugin/dist/loader.js',
       [Object],
       [Object] 
    ] 
}, {
    loader: 'url-loader',
    exclude: [ /\.js$/, /\.html$/, /\.json$/ ],
    query: { limit: 10000, name: 'static/[name].[hash:8].[ext]' } 
}]

stage:  node
config.module.rules[1].oneOf:  [{ 
    test: /\.(js|jsx)$/,
    exclude: /(node_modules|\/Users\/justfly\/projects\/aggrow-static\/node_modules)/,
    use: [ [Object] ] 
}, { 
    test: /\.css$/, 
    loader: [ [Object], [Object] ] 
}, { 
    loader: 'url-loader',
    exclude: [ /\.js$/, /\.html$/, /\.json$/ ],
    query: { limit: 10000, name: 'static/[name].[hash:8].[ext]' } 
}]
```

For me it is obviously missing '/Users/justfly/projects/aggrow-static/node_modules/extract-css-chunks-webpack-plugin/dist/loader.js' in node webpack.config.js

This is the same problem I had since react-static@5.8.2 was released. I think `extract-css-chunks-webpack-plugin` was removed from node webpack config, cos it was generating second css file, but it is crucial for `CSS modules`

currently I'm stuck - I've added `extract-css-chunks-webpack-plugin` to config, I'm getting css file, but elements in static html files are without class attributes. Hope will fix that till the end of the week and post my webpack config here ASAP.

**Comment by jnewc** [comment]
This still appears to be broken due to the sass plugin example still being non-functional. I've tried adding the plugin to a fresh project but, after adding some debug logs, it is clear that it's never loaded/executed.

Can this be reopened?

**Comment by nozzle** [comment]
Info: `master` branch now points to 5.x.x and the `v6` branch to 6.0.0-beta.

**Comment by ScriptedAlchemy** [comment]
Update: there is a PR open which will resolve the second half of the problem, originating from some issues i inherited from webpacks work on mini-css

Can confirm the build is working now

https://github.com/nozzle/react-static/pull/733

**Comment by nikaspran** [comment]
@tannerlinsley @ScriptedAlchemy Please correct me if I'm wrong on all of this, but it's late and I've been looking at the internals of my node_modules for too long to maintain any sanity at this point. This is going to be a long one.

This issue should be reopened - issue still persists in both 5.x and 6.x branches. The PR that broke CSS modules is indeed https://github.com/nozzle/react-static/pull/574. Explanation following below, but first:

# **TL;DR: Workaround until there's an official fix:**

## v5.x
* Adjust your custom webpack config to always use `ExtractCssChunks` or `ExtractTextPlugin` and always add the plugin:
  ```diff
  -// Don't extract css to file during node build process
  -if(stage !== 'node') {
    loaders = ExtractCssChunks.extract({
      fallback: {
        loader: 'style-loader',
        options: {
          sourceMap: false,
          hmr: false,
        },
      },
      use: loaders,
    })
  -}

  +if (stage === 'node') {
  +  config.plugins.push(new ExtractCssChunks());
  +}
  ```
* Same strategy seems to work for both `ExtractTextPlugin` and `ExtractCssChunks`, just different APIs obviously - make sure to tailor this to your webpack config.

## v6.x
* Adjust your custom webpack config to always use `ExtractCssChunks` and always add the plugin:
  ```diff
  if (stage === 'dev') {
    // Dev
    loaders = [styleLoader, cssLoader, sassLoader]
  } else if (stage === 'node') {
    // Node
  - // Don't extract css to file during node build process
  - loaders = [cssLoader, sassLoader]
  + loaders = [ExtractCssChunks.loader, cssLoader, sassLoader]
  + config.plugins.push(new ExtractCssChunks({
  +   filename: '[name].[chunkHash:8].css',
  +   chunkFilename: '[id].[chunkHash:8].css',
  + }));
  } else {
    // Prod
    loaders = [ExtractCssChunks.loader, cssLoader, sassLoader]
  }

  config.module.rules[0].oneOf.unshift({
    test: /\.s(a|c)ss$/,
    use: loaders,
  })
  ```

## Something else?

* If your configuration is significantly different from the example SASS ones, the same concepts should apply but you might want to read the details below to figure out how to fix it for yourself.

# Explanation & details

Here's how the CSS module pipeline usually works for JSX:

Say you have these two files (I'm using SCSS to use an unrelated fresh asset pipeline, but the same concepts apply if replacing the default css loader with a css modules enabled one manually):

```scss
// Home.scss
.Home {
  color: red;
}
```

```jsx
// Home.js
import React from 'react';
import styles from './Home.scss';

console.log(styles);

export default function Home() {
  return (
    <div className={styles.Home}>
      Home
    </div>
  );
}
```

This is the configuration we'll be using (slightly adjusted from the sass example one). **NOTE**: the only difference is the ` modules: true,` option on `css-loader`:

```js
webpack: (config, { defaultLoaders, stage }) => {
    let loaders = []

    if (stage === 'dev') {
      loaders = [{ loader: 'style-loader' }, { loader: 'css-loader' }, { loader: 'sass-loader' }]
    } else {
      loaders = [
        {
          loader: 'css-loader',
          options: {
            importLoaders: 1,
            minimize: stage === 'prod',
            sourceMap: false,
            modules: true, // <- the only difference is here
          },
        },
        {
          loader: 'sass-loader',
          options: { includePaths: ['src/'] },
        },
      ]

      if (stage !== 'node') {
        loaders = ExtractCssChunks.extract({
          fallback: {
            loader: 'style-loader',
            options: {
              sourceMap: false,
              hmr: false,
            },
          },
          use: loaders,
        })
      }
    }

    config.module.rules = [
      {
        oneOf: [
          {
            test: /\.s(a|c)ss$/,
            use: loaders,
          },
          defaultLoaders.cssLoader,
          defaultLoaders.jsLoader,
          defaultLoaders.fileLoader,
        ],
      },
    ]
    return config
  },
```

## Before https://github.com/nozzle/react-static/pull/574

For 5.x, using the example configuration:
1. Use `css-loader` with `modules: true` option, webpack will generate the following webpack result module:
```
/***/ }),
/* 133 */
/***/ (function(module, exports, __webpack_require__) {

exports = module.exports = __webpack_require__(134)(false);
// imports


// module
exports.push([module.i, "._1KGCDNoWpJiH3oe8eoX-{color:red}", ""]);

// exports
exports.locals = {
	"Home": "_1KGCDNoWpJiH3oe8eoX-"
};
```
2. Use `ExtractTextPlugin` or `ExtractChunksPlugin` to get the `exports.locals` directly. If you add that plugin, this will result in the following:
```
/***/ }),
/* 133 */
/***/ (function(module, exports) {

// removed by extract-text-webpack-plugin
module.exports = {"Home":"_1KGCDNoWpJiH3oe8eoX-"};

/***/ })
```
3. We can see the correct exports have been produced and CSS modules work as intended - perfect. This is how it used to work before https://github.com/nozzle/react-static/pull/574. The only side effect is we're building more css files than needed since we're running this plugin twice.

## After https://github.com/nozzle/react-static/pull/574

By not including `ExtractTextPlugin` or `ExtractCssChunks` in the `node` stage, during bundle step the `static.[hash].js` file does **NOT** have the `ExtractTextPlugin` or `ExtractChunksPlugin` applied and we're stuck with the result from Step 1 above - hence css modules don't work. Note that the `prod` step result still looks and works fine but since the HTML is prebuilt using the `node` step output, the result is that none of the class names are applied (they're all accessible to components via `styles.locals.<className>` rather than `styles.<className>` 

## 6.x ?

There's essentially the same problem, except we're using newer versions of everything so the APIs are a bit different. Core of the issue is still the same - we're not using `ExtractCssChunks` plugin for the `node` stage which means that css modules built with `css-loader` are not getting properly transformed for HTML generation.

# Solutions?

Considering we don't want to be building duplicate CSS files during the node step, we still need a way to generate and extract the correct locals for CSS modules. I've posted workarounds at the top but those are fairly dirty and might break when updating React Static. The way I can see it there are a few options here:
1. Add ExtractCssChunks plugin back into the plugins for the `node` stage. This seems to work fine and the only issue I could find is the unused css files.
1. Add a "dry-run" config option to ExtractCssChunks that would just transform the webpack modules but not actually output any CSS. Seems to be going slightly against the idea since the plugin is not actually necessary for this.
1. Create a lightweight loader to transform the css-loader result into just the locals - kinda like `style-loader` but without the actual injecting into the DOM, just this transform basically:

  ```js
  if(content.locals) module.exports = content.locals;
  ```

# CSS modules and React Static

On a side note, considering the adoption rate of CSS modules, it would be great to have more support for them. I can see two options here and I'm willing to work on both:
1. Support CSS modules out of the box via a configuration option
2. Add an example for a custom webpack config to add the support

Which one would align better with the direction of React Static @tannerlinsley?

**Comment by lottamus** [comment]
Let’s add it back. Outputting that file isn’t expensive and we could add a step to delete the file or leave it up to the end user to delete.

**Comment by ScriptedAlchemy** [comment]
Also referring to the other comment, in an ideal situation - extract css chunks should usually be run, for client builds. On the server, I typically use the following configuration 

https://github.com/faceyspacey/universal-demo/blob/master/webpack/server.dev.js

**Comment by lz-allen** [comment]
@Zack Jackson
I don't know why I always build error when I use "extract-css-chunks-webpack-plugin", on the contrary, "extract-text-webpack-plugin" won't.    But the className built by this plug-in is undefined
I also purposely upgraded webpack 4.0

**Comment by ScriptedAlchemy** [comment]
Okay this is the same issue as some others are having.  I patch it directly in projects. Here is it.

It was literally exactly this where stuff comes back undefined. 

![50781514-1f30-4012-a57c-0879c26a73fb](https://user-images.githubusercontent.com/25274700/49701964-bdf72100-fba7-11e8-82ed-9a3df27efbeb.jpeg)

**Comment by ScriptedAlchemy** [comment]
@tannerlinsley do we have a cssModules option currently exposed to the user?

**Comment by nikaspran** [comment]
@ScriptedAlchemy please see https://github.com/nozzle/react-static/issues/601#issuecomment-429574588 - this has nothing to do with ExtractCssModules itself, react-static just does not include it in the `node` stage and thus the classes never get exported as properties for css module imports.

I still haven't gotten around to creating a PR, but the workaround definitely works for both 5.x and 6.x.

**Comment by ScriptedAlchemy** [comment]
Ohhhh I noticed this before. It’s not emitting it the same way or something? I’ve been mening to look into it. please tag myself in the pr when you have time. It would be a huge help to many!

**Comment by JustFly1984** [comment]
Guys don't waste your time with react-static, I've migrated every project I had to gatsby, and gatsby is faster, reliable and does support css modules out of the box without the need to hustle with webpack config, although you can do that too.

**Comment by Pixelatex** [comment]
@JustFly1984 We are not gatsby and if you're not helping, please don't be an ass about it as well.

**Comment by JustFly1984** [comment]
@Pixelatex I've been trying to work my way with react-static and CSS modules for several months, and there is too much issues among the webpack configuration. At the end you get extra css file embedded to <head> and you can't do nothing about it.

**Comment by ScriptedAlchemy** [comment]
Not the culture I’d like to see on the project, especially since gatsby is a funded project with millions poured into it. 

I understand your frustration, and I do know that RS does indeed have complex Webpack composers which do make it hard to deal with build problems. But please understand RS6 is still in beta. 

I understand your desire to move from frustration, but authors are under strain and there’s not that many of us. We do our best where we can. Trashing a project in beta is disrespectful. 

Your opinion matters and your frustration is completely valid. But please don’t slam react static because of a css bug. 

All in all, it took me seconds to rectify it in the projects of those who reached out for support. 
Maybe a fix isn’t released, which I’ll follow up on. But damn, I get you’re upset, but that’s cold. 

I’m confident if you can arrange $500k, we could compete with gatsby given a fraction of the budget and resources. That’s beyond the point though.

**Comment by JustFly1984** [comment]
It does not support CSS modules, and it will not support it for long time. This is a thread about CSS modules. I'm telling everybody who does need CSS modules as I am that it does not resolving this issue since 5.8.2 version. I was pointing this out for long time, and wasted my time.

**Comment by Pixelatex** [comment]
@ScriptedAlchemy Agreed, let's get this thing back on topic and fix it for the people who care

**Comment by ScriptedAlchemy** [comment]
Actually, I reached out and personally pull requested your private project.  My PR remains open, it was you who never responded to me. 

I will not let this ruin my day. But I cared about your challenges and even though I was exhausted, I went out of my way to fix your project free of charge. You could have voiced your opinion in a less emotional manner and / or pr worked with us. 

Screw me tho, right?

![29ee122d-37c1-48f9-b761-206d43438b12](https://user-images.githubusercontent.com/25274700/49703879-872d0500-fbbf-11e8-9395-d17e40b8c3be.png)

**Comment by JustFly1984** [comment]
This did not worked out, I had the same issue after your fixes. I do not want to ruin anybody's day.

**Comment by ScriptedAlchemy** [comment]
Then give me some feedback dude! You had the author of half this crap litter all working on your private project for free. I was waiting for your review so I could fix the issue once and for all. The fix I was trying to apply for you would have been the PR I open to this project. 

Emotions aside. I do apologize that we didn’t pull through for you. If you do end up using either my tools or ones I’m accociated with. **feel free to reach out, I am always willing to support our users and care about the success of their projects**

**Comment by ScriptedAlchemy** [comment]
Now back to the point. @nikaspran, if we just include css plugin in node, will that rectify the issue based on your experience. 

Would you be willing do join my slack channel and over a few messages one evening hammer this out together. I myself do not have a project using RS with css modules. 

I’ve got write access, but help me pinpoint it.. what did you do to fix it? I’ll open a pr today if you have a demo project you can share, and are willing to speak to me

**Comment by ScriptedAlchemy** [comment]
Compared against Icarus, this should be what we need in terms of how css module should work on node

https://github.com/ScriptedAlchemy/Icarus/blob/master/build/webpack.config.js/loaders.js

**Comment by ScriptedAlchemy** [comment]
Okay this is actually simple to fix. 

I’m going to add a new example project to the generator, then go back and write a comment along with the code in the sass examples and such that says “in-comment  this if you want modules”

**Comment by tannerlinsley** [comment]
Sounds like things got a little heated for a bit. Given the conversation I'd like to echo and clarify some things from our contribution guidelines:

- This is open source, not a business, but that doesn't mean it's free. It's an exchange of knowledge and support to make great tools accessible to the world. 
- React Static is developed and maintained for free by a number of very generous people. Thank you all for the help!
- GitHub and the contributor slack channel are not a support network. Keep support and questions of implementation on Spectrum.
- Competition is good, but trash talking a library you do not pay for and are not willing to actively contribute to is considered very disrespectful in my book and won't be tolerated in any of my repos. 
- Let's make some awesome software!

Carry on!

**Comment by JustFly1984** [comment]
@tannerlinsley sorry bro, I didn't mean to be considered trash talking. Peace everybody.

**Comment by ScriptedAlchemy** [comment]
PR is open.  Waiting on Tanner so i know how to test it via RS CLI, and obviously, once I'm at a computer. If someone can assist in testing it, ill be much obliged. 

@nikaspran thank you very much for your efforts in debugging this. While i own some of the underlying dependencies, there are areas of RS i am not familiar with. 

I again apologize for the inconvenience this issue has caused all of our users, let us work together in a constructive manner and put CSS modules to bed once and for all. Furthermore, for the heated discussion that has likely spammed innocent bystanders in this thread. 

@nikaspran @Pixelatex @lz-allen @Cinamonas, and all others. I will make myself available every night this week to resolve any outstanding issues regarding CSS modules.

**Comment by ScriptedAlchemy** [comment]
@nikaspran new PR opened

Let me know if you have any other contributions.

https://github.com/nozzle/react-static/pull/870

**Comment by tannerlinsley** [comment]
With #870 merged, we can now put this to rest 🎉

**Comment by ScriptedAlchemy** [comment]
anyone who experiences any more issues, please tag @ScriptedAlchemy - Ive tried to cover the scenarios. Remember, im not perfect and usually screw things up the first 2 times 🤣. 

Im more than happy to take on challanges and of course fix css modules issues.
Lets all be positive, if you are in frustrating situations - we tottaly empathise!

Thank you all for the patience especially since the issue was opened in **May** 

100% my fault for not stepping in and resolving the issue. No excuses
